.github/workflows/java.yml

```diff
@@ -26,47 +26,76 @@ on:
             - .github/json_matrices/build-matrix.json
     workflow_dispatch:
 
+    pull_request_target:
+        types: [labeled]
+        paths:
+            - glide-core/src/**
+            - submodules/**
+            - java/**
+            - .github/workflows/java.yml
+            - .github/workflows/install-shared-dependencies/action.yml
+            - .github/workflows/test-benchmark/action.yml
+            - .github/workflows/lint-rust/action.yml
+            - .github/workflows/install-valkey/action.yml
+            - .github/json_matrices/build-matrix.json
+    schedule:
+        - cron: "0 0 * * *"
+
 concurrency:
     group: java-${{ github.head_ref || github.ref }}
     cancel-in-progress: true
 
 jobs:
-    load-engine-matrix:
+    get-matrices:
         runs-on: ubuntu-latest
         outputs:
-            matrix: ${{ steps.load-engine-matrix.outputs.matrix }}
+            should-run: ${{ steps.get-matrices.outputs.should-run }}
+            engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}
+            host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}
+        steps:
+            - uses: actions/checkout@v4
+            - id: get-matrices
+              uses: ./.github/workflows/create-tests-matrices
+              with:
+                  language-name: java
+            - name: Echo outputs
+              run: |
+                  echo "should-run: ${{ steps.get-matrices.outputs.should-run }}"
+                  echo "engine-matrix-output: ${{ steps.get-matrices.outputs.engine-matrix-output }}"
+                  echo "host-matrix-output: ${{ steps.get-matrices.outputs.host-matrix-output }}"
+    create-version-matrix:
+        runs-on: ubuntu-latest
+        needs: get-matrices
+        if: ${{ needs.get-matrices.outputs.should-run == 'true' }}
+        outputs:
+            java-matrix-output: ${{ steps.create-version-matrix.outputs.java-matrix-output }}
         steps:
             - name: Checkout
               uses: actions/checkout@v4
```

Comment:
I think we don't need checkout in this step

Reply:
if matrices files changes we do need it